### DISCUSSION TITLE
What is the best strategy to delete data from two related entities?

### DISCUSSION THREAD

**Discussion by kmrozConfigit** [ravendb/ravendb#15472] [Q&A] [answered]
Hi, I need to ask for advice. Want to find the best solution for deleting data from two related entities (many rows). 
1. have e.g. http://live-test.ravendb.net/studio/index.html#databases/documents?collection=Orders&database=demo-aviv
2. have Orders 
{ "Company": "companies/65-A",
    "Employee": "employees/1-A",
...}
4. and have related Companies 
{ "ExternalId": "NORTS",
    "Name": "North/South",
   ...
...},
{"ExternalId": "WARTH",
    "Name": "Wartian Herkku",
...}

I can get them, load etc., but what would be the best way to delete all Orders for company Name = "North/South" and Company? 
(I'd like to use RQL Patch or node.js script by API, and the plan is to do it regularly/automatically to delete old data by creation date)

**Comment by ayende** [comment, marked as answer]
```
from Orders
where Company = 'companies/1-A'  // filter
update {
   del(this.Employee); // del related
   del(id(this)); // del order
}
```

**Reply by kmrozConfigit** [reply]
I decided finally use combination of techniques (DocumentStore and Curl) to delete old Documents and related Logs:
```
...
do {
        results = await session
            .query({ collection: 'Documents' })
            .whereLessThan('CreatedDate', firstDayOfTheYear)
            .statistics(s => stats = s)
            .skip((pageNumber * PAGE_SIZE) + skippedResults)
            .take(PAGE_SIZE)
            .all();

        if (results. Length > 0) {
            const where = "'" + results. Map(x => session.advanced.getDocumentId(x)).join(`','`) + "'";           
            query = `{ "Query": "from Logs where DocumentId in (${where})" }`;            
            xhr.open("DELETE", url);
            xhr.send(query);
        }

        skippedResults += stats.skippedResults;
        pageNumber++;
    } while (results. Length > 0);

    query = `{ "Query": "from Documents where CreatedDate < '${firstDayOfTheYear.toISOString()}'" }`;    
    xhr.open("DELETE", url);
    xhr.send(query);
...
```
